StRoot/StFcsDbMaker/StFcsDbPulse.cxx

```diff
@@ -0,0 +1,185 @@
+#include "StFcsDbPulse.h"
+
+ClassImp(StFcsDbPulse);
+
+StFcsDbPulse::StFcsDbPulse(const char* name) : TDataSet(name)
+{
+  mTail = 0;
+  mTBPerRC = 8;
+
+  mGSigma = 0;
+  mA1     = 0;
+  mA2     = 0;
+  mXoff1  = 0;
+  mXoff2  = 0;
+  mTau1   = 0;
+  mTau2   = 0;
+  mP1     = 0;
+  mP2     = 0;
+}
+
+StFcsDbPulse::~StFcsDbPulse(){}
+
+int StFcsDbPulse::Init()
+{
+
+  mGSigma = 24.5;    // signal nominal sigma
+  //GSigmaSigma = 0;  // distibution of signal sigma
+  mA1    = 1.0/sqrtpi()/0.155/129;
+  mA2    = 0.2/sqrtpi()/0.155/129;
+  mXoff1 = 70 - 129;
+  mXoff2 = 220 - 129;
+  mTau1  = 200.0;
+  mTau2  = 40.0;
+  mP1    = 1.0;
+  mP2    = 1.0;
+
+  return 0;
+}
+
+void StFcsDbPulse::setTail(int tail)
+{
+  mTail = tail;
+  if( tail == 1 )
+    {
+      //Data from Gerard 2020 summer
+      mGSigma    = 24.5/nsecPerTB();
+      mA1        = 1.0/0.154/mGSigma;
+      mA2        = 0.2/0.154/mGSigma;
+      mXoff1     = (70 - 129)/nsecPerTB();
+      mXoff2     = (220 - 129)/nsecPerTB();
+      mTau1      = 200.0/nsecPerTB();
+      mTau2      = 40.0/nsecPerTB();
+      mP1        = 1.0;
+      mP2        = 1.0;
+    }
+  else if( tail == 2 )
+    {
+      //Data from WAH with real detector/LED system 2021 Jan
+      mGSigma    = 2.347;
+      mA1        = 2543.0/854.0/mGSigma;
+      mA2        = 0.0;
+      mXoff1     = 211.3-215.7;
+      mXoff2     = 0.0;
+      mTau1      = 4.375;
+      mTau2      = 0.0;
+      mP1        = 1.0;
+      mP2        = 0.0;
+    }
+  else{ std::cout << "WARNING:Invalid Tail value" << std::endl; }
```

Comment:
See https://star-bnl.github.io/star-coding-guidelines/formatguide.xml#Printing_Messages_in_STAR

Reply:
Which header do I need to be able to use LOG_XXX?